core/src/main/java/org/infinispan/container/offheap/OffHeapDataContainer.java

```diff
@@ -202,7 +205,6 @@ protected void performPut(long address, long newAddress, WrappedBytes key) {
             if (!foundKey) {
                if (offHeapEntryFactory.equalsKey(address, key)) {
                   entryReplaced(newAddress, address);
-                  allocator.deallocate(address);
```

Comment:
I would like to investigate this further as well. My initial thoughts was this was possible, however I found an issue, that I cannot remember now. Instead I implemented it this way for right now, since I know for sure that it will work.

I created https://issues.jboss.org/browse/ISPN-8437 to do this later though.

Reply:
Ok